src/bdf_utils.jl

```diff
@@ -23,3 +23,19 @@ function R!(k, ρ, cache)
     end
   end
 end
+
+function backward_diff(D, D2, k, flag=true)
```

Comment:
I did this flag trick for a particular case of order 5, `EEst` at order `k+1=5+1=6` will be required for order conditions hence we need `del^(6)y_n`. Size of `D` is 5, so can't store it (also we don't need to store it, we can access it by `D2[6,1]`) and If we make the size of `D` to 6, it will create a problem for matrix multiplication. Got it?

Reply:
> I don't like the name D and D2

could you please suggest name